### PR TITLE
Error instead of panic on duplicat loca entry

### PR DESCRIPTION
**Pull request by jschwe**
This brings the rust implementation in-line with the reference implementation which also errors cleanly.

Note: The issue was found during AI review of the code and the fix and test largely generated by an agent.
Existing tests continue to pass.

### CODE REVIEW

**Review by nicoburns — COMMENTED** [review]
A couple of thoughts:

1. Do you have a way of visualising the font file? I'm a little reluctant to commit AI-generated binary code that I can't review! (although it's *technically* probably ok here, as we are expecting to deal with untrusted input)

2. I think we ought to try to upstream these test cases to https://github.com/web-platform-tests/wpt/tree/master/css/WOFF2. There is a copy of those fonts in this repo under conformance/wpt/woff2. The WPT tests have HTML/CSS wrappers which generate a layout which depends on either the font loading or the font not loading. But we just assert byte-identical output with the C++ decoder.

**Comment by jschwe** [comment]
> I'm a little reluctant to commit AI-generated binary code that I can't review! (although it's technically probably ok here, as we are expecting to deal with untrusted input)

Thats fair. 

```
diff conformance/wpt/woff2/directory-knowntags-001.woff2 wuff/tests/fixtures/valid-transformed-loca.woff2 
```

These files are identical (as the comment in the test-cases says), so I guess we could also directly use it via a relative path instead of vendoring it again. 

The other fixture is derived from that you can inspect by comparing with the woff2_info command, which is built via cmake already (if you ran the conformance tests).

```
diff -u <(woff2/build/woff2_info wuff/tests/fixtures/valid-transformed-loca.woff2) <(woff2/build/woff2_info wuff/tests/fixtures/duplicate-loca.woff2)
--- /dev/fd/63  2026-07-15 09:43:20.182881644 +0000
+++ /dev/fd/62  2026-07-15 09:43:20.182881644 +0000
@@ -1,9 +1,9 @@
-Processing wuff/tests/fixtures/valid-transformed-loca.woff2 => wuff/tests/fixtures/valid-transformed-loca.woff2
+Processing wuff/tests/fixtures/duplicate-loca.woff2 => wuff/tests/fixtures/duplicate-loca.woff2
 WOFF2Header
 signature           0x774f4632
 flavor              0x00010000
-length              1504
-numTables           11
+length              1508
+numTables           12
 reserved            0
 totalSfntSize       3616
 totalCompressedSize 1416
@@ -24,7 +24,8 @@
     5     67  0x02 hhea         36
     6     69  0x43 hmtx         16        9
     7     72  0x0b loca         10        0
-    8     75  0x04 maxp         32
-    9     77  0x3f name        621
-   10     84  0x07 post         32
-TableDirectory ends at +86
+    8     75  0x0b loca         10        0
+    9     78  0x04 maxp         32
+   10     80  0x3f name        621
+   11     87  0x07 post         32
+TableDirectory ends at +89
```

But I guess upstreaming the test might be simpler.

**Comment by jschwe** [comment]
I'd prefer not having main being force-pushed, perhaps you could disable that in the branch protections of this repo?

**Comment by nicoburns** [comment]
> I'd prefer not having main being force-pushed, perhaps you could disable that in the branch protections of this repo?

That's fair. ~~Will do~~ Done.

**Comment by nicoburns** [comment]
I've manually merged a variant of this that:

- (includes your fix unmodifed)
- Updates the `conformance` test runner to count panics as failures
- Adds the test font you generated to `conformance/fonts/custom` which gets picked up as a test font

**Comment by jschwe** [comment]
Regarding upstreaming the test to wpt (which in turn comes from https://github.com/w3c/woff2-tests) - it seems that there is no requirement in the spec for the decoder to reject a second TableDirectoryEntry (there is a requirement for encoders to emit only a single entry), but what a decoder should do is unspecified (and e.g. ignoring it might also be a spec-conform solution).
So I'm not sure if it makes sense trying to upstream this test.

**Comment by nicoburns** [comment]
> Regarding upstreaming the test to wpt (which in turn comes from https://github.com/w3c/woff2-tests) - it seems that there is no requirement in the spec

Yeah, we're definitely starting to get into run into edge cases where the upstream C++ decoder's behaviour doesn't necessarily match the spec. At some point we may have to account for cases where it's the C++ decoder that's wrong and we want to implement something else (although in this case the behaviour seems fine).
